implementation/src/main/java/io/smallrye/jwt/config/JWTAuthContextInfoProvider.java

```diff
@@ -120,9 +121,30 @@ private static JWTAuthContextInfoProvider create(String publicKey, String public
     @Inject
     @ConfigProperty(name = "smallrye.jwt.token.cookie")
     private Optional<String> tokenCookie;
-
     /**
-     * Default group name. This property can be used to support the JWT tokens without a 'groups' claim.
+     * Check that the JWT has at least one of 'sub', 'upn' or 'preferred_user_name' set. If not the JWT validation will
+     * fail. If the properties 'smallrye.jwt.claims.sub' or 'smallrye.jwt.sub.path' are set it will use these values
+     * to validate instead.
+     */
+    @Inject
+    @ConfigProperty(name = "smallrye.jwt.require.named-principal", defaultValue = "true")
```

Comment:
Hi @radcortez While I'm chasing the issue at the Thorntail side, can you please change the default value to `false` as indeed there could be some custom tokens around without those 3 claims. One would use `true` only to ensure that the application code can always rely on Java principal name

Reply:
Ok.

Reply:
@sberyozkin done!

Reply:
Thanks; one more thing, please update the docs of this property, remove the 2nd sentence starting from 'If'. And after you also update the whitelist property to Optional I'll be ready to merge I hope :-)

Reply:
Done :)